### DISCUSSION TITLE
Depending on another addon

### DISCUSSION THREAD

**Discussion by Bilal2453** [LuaLS/LLS-Addons#44] [Ideas]
One of the possible ways of cutting the size of an addon off is having less redundant definitions, for example the Luvit addon also defines definitions for lpeg, re module of lpeg, dkjson, luv, all of which are general packages that can be used in normal Lua.

Another advantage to being able to depend on another addon is that you you no more have to redefine them, making maintenance and updates easier and faster.

Therefor, what are your thoughts on implementing a dependency system?  It should be pretty easy to implement from what I can notice.

**Comment by carsakiller** [comment]
I have been wanting to add in a way for addons to depend on each other, however, I think there are even larger changes to make. I would like to find a way for everything to work across the various platforms and editors. Ultimately, this likely comes down to the addon manager being moved into the language server itself, rather than in the VS Code extension. This will require a lot of research and work on my part, though, which I just don't have the time for just yet.
